homeassistant/components/frontend/__init__.py

```diff
@@ -281,22 +292,50 @@ def async_setup(hass, config):
 
     repo_path = conf.get(CONF_FRONTEND_REPO)
     is_dev = repo_path is not None
+    hass.data[DATA_JS_VERSION] = js_version = conf.get(CONF_JS_VERSION)
 
     if is_dev:
         hass.http.register_static_path(
             "/home-assistant-polymer", repo_path, False)
         hass.http.register_static_path(
             "/static/translations",
-            os.path.join(repo_path, "build/translations"), False)
-        sw_path = os.path.join(repo_path, "build/service_worker.js")
+            os.path.join(repo_path, "build-translations"), False)
+        sw_path_es5 = os.path.join(repo_path, "build-es5/service_worker.js")
+        sw_path_latest = os.path.join(repo_path, "build/service_worker.js")
         static_path = os.path.join(repo_path, 'hass_frontend')
     else:
         import hass_frontend
-        frontend_path = hass_frontend.where()
-        sw_path = os.path.join(frontend_path, "service_worker.js")
-        static_path = frontend_path
+        import hass_frontend_es5
+        sw_path_es5 = os.path.join(hass_frontend_es5.where(),
+                                   "service_worker.js")
+        sw_path_latest = os.path.join(hass_frontend.where(),
+                                      "service_worker.js")
+        # /static points to latest dir. However all files that differ between
```

Comment:
While testing this, I got confused by if my routes are now using ES5 or not. All of a sudden realized how we can make this a lot easier. Instead of depending on route overrides, wouldn't it be better if we have something like

- `/static` for truly static assets, like icons, images
- `/app` for untranspiled application assets
- `/app_es5` for ES5 application assets

Updating the frontend to do this shouldn't be hard. We just don't have to copy the static files into the final build dir but instead be their own dir.

It would untangle a lot of this.

Reply:
Will do.

Reply:
Done: `frontend_latest` and `frontend_es5`
aiohttp.StaticResource is per prefix and not per dir, so having `/frontend/` and `/frontend_es5/` confuses it.